includes/apple-exporter/builders/class-components.php

```diff
@@ -1,486 +1,708 @@
 
-		// Regardless of what the component class specifies,
-		// add the targetComponentIdentifier here.
-		// There's no way for the class to know what this is before this point.
+		// Regardless of what the component class specifies, add the
+		// targetComponentIdentifier here. There's no way for the class to know what
+		// this is before this point.
 		$anchor_json['targetComponentIdentifier'] = $target_component->uid();
 
-		// Add the JSON back to the component
+		// Add the JSON back to the component.
 		$component->set_json( 'anchor', $anchor_json );
 
 		// Given $component, find out the opposite position.
-		$other_position = null;
-		if ( Component::ANCHOR_AUTO == $component->get_anchor_position() ) {
-			$other_position = 'left' == $this->get_setting( 'body_orientation' ) ? Component::ANCHOR_LEFT : Component::ANCHOR_RIGHT;
-		} else {
-			$other_position = Component::ANCHOR_LEFT == $component->get_anchor_position() ? Component::ANCHOR_RIGHT : Component::ANCHOR_LEFT;
+		$other_position = Component::ANCHOR_LEFT;
+		if ( ( Component::ANCHOR_AUTO === $component->get_anchor_position()
+		       && 'left' !== $this->get_setting( 'body_orientation' ) )
+		     || Component::ANCHOR_LEFT == $component->get_anchor_position()
+		) {
+			$other_position = Component::ANCHOR_RIGHT;
 		}
-		$target_component->set_anchor_position( $other_position );
+
 		// The anchor method adds the required layout, thus making the actual
 		// anchoring. This must be called after using the UID, because we need to
 		// distinguish target components from anchor ones and components with
 		// UIDs are always anchor targets.
+		$target_component->set_anchor_position( $other_position );
 		$target_component->anchor();
 		$component->anchor();
 	}
 
 	/**
-	 * Add a thumbnail if needed.
+	 * Estimates the number of chars in a line of text next to an anchored component.
+	 *
+	 * @since 1.2.1
 	 *
-	 * @param array &$components
 	 * @access private
+	 * @return int The estimated number of characters per line.
 	 */
-	private function add_thumbnail_if_needed( &$components ) {
-		// If a thumbnail is already defined, just return.
-		if ( $this->content_cover() ) {
-			return;
-		}
+	private function _characters_per_line_anchored() {
 
-		// Otherwise, iterate over the components and look for the first image.
-		foreach ( $components as $i => $component ) {
-			if ( is_a( $component, 'Apple_Exporter\Components\Image' ) ) {
-				// Get the bundle URL of this class.
-				$json_url = $component->get_json( 'URL' );
-				if ( empty( $json_url ) ) {
-					$json_components = $component->get_json( 'components' );
-					if ( ! empty( $json_components[0]['URL'] ) ) {
-						$json_url = $json_components[0]['URL'];
-					}
-				}
+		// Get the body text size in points.
+		$body_size = $this->get_setting( 'body_size' );
 
-				if ( empty( $json_url ) ) {
-					return;
-				}
+		// Calculate the base estimated characters per line.
+		$cpl = 20 + 230 * pow( M_E, - 0.144 * $body_size );
```

Comment:
I used a curve fitting algorithm for this, so yeah, it is a bit of a mess, but it works :)